### PR TITLE
ZEK-1892-FE-FRC-Fix-Prevent-Saving-if-not-valid-code

### PR DESCRIPTION
**Pull request by Yusuf-00**
- added get annotations of the code editor 
- added handle error method and emit function on error to be emitted if there is errors

### CODE REVIEW

**Review by AbdAsh — DISMISSED** [review]
LGTM

File: src/components/code-editor/CodeEditor.vue [hunk 1]
```diff
@@ -104,6 +105,13 @@ export default {
                 }
             });
         },
+        handleError(annotations) {
+            // Filter annotations with type 'error'
+            const errorAnnotations = annotations.filter(annotation => annotation.type === "error");
+
+            // Emit 'onError' event with the array of error annotations
+            this.$emit("onError", errorAnnotations);
```
Comment (AbdAsh) [inline src/components/code-editor/CodeEditor.vue:113]: The `onError` event should only be emitted if there are errors

**Review by AbdAsh — CHANGES_REQUESTED** [review]

File: src/components/code-editor/CodeEditor.vue (hunk 1 above)
Comment (Yusuf-00) [reply src/components/code-editor/CodeEditor.vue:113]: will add the check for it back @AbdAsh

Comment (Yusuf-00) [reply src/components/code-editor/CodeEditor.vue:113]: sent replay regarding this on slack

**Review by AbdAsh — APPROVED** [review]
NP LGTM

**Pull request merged**